nifi-nar-bundles/nifi-apicurio-bundle/nifi-apicurio-schema-registry-service/src/main/java/org/apache/nifi/apicurio/schemaregistry/ApicurioSchemaRegistry.java

```diff
@@ -0,0 +1,135 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.apicurio.schemaregistry;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnEnabled;
+import org.apache.nifi.apicurio.schemaregistry.client.ApicurioSchemaRegistryClient;
+import org.apache.nifi.apicurio.schemaregistry.client.CachingSchemaRegistryClient;
+import org.apache.nifi.apicurio.schemaregistry.client.SchemaRegistryApiClient;
+import org.apache.nifi.apicurio.schemaregistry.client.SchemaRegistryClient;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.schema.access.SchemaField;
+import org.apache.nifi.schema.access.SchemaNotFoundException;
+import org.apache.nifi.schemaregistry.services.SchemaRegistry;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.serialization.record.SchemaIdentifier;
+import org.apache.nifi.web.client.provider.api.WebClientServiceProvider;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@Tags({"schema", "registry", "apicurio", "avro"})
+@CapabilityDescription("Provides a Schema Registry that interacts with the Apicurio Schema Registry so that those Schemas that are stored in the Apicurio Schema "
+        + "Registry can be used in NiFi. When a Schema is looked up by name by this registry, it will find a Schema in the Apicurio Schema Registry with their names.")
+public class ApicurioSchemaRegistry extends AbstractControllerService implements SchemaRegistry {
+
+    private static final Set<SchemaField> schemaFields = EnumSet.of(SchemaField.SCHEMA_NAME, SchemaField.SCHEMA_TEXT,
+            SchemaField.SCHEMA_TEXT_FORMAT, SchemaField.SCHEMA_IDENTIFIER, SchemaField.SCHEMA_VERSION);
+
+    static final PropertyDescriptor SCHEMA_REGISTRY_URL = new PropertyDescriptor.Builder()
+            .name("schema-registry-url")
+            .displayName("Schema Registry URL")
```

Comment:
Following other more recent conventional changes, the name and displayName should be the same.

Reply:
Where does this new change come from?

Reply:
Good question, the recent addition of [NIFI-12139](https://issues.apache.org/jira/browse/NIFI-12139) provides new API methods to migrate existing property names to new property names. Part of this capability provides the opportunity to unify the property name and display name, which makes it more straightforward to translate component configuration through the UI to programmatic calls. Previously, the displayName provided a way to rename properties without breaking existing flows, but with the new migrate properties capability, that will no longer be necessary. For new components, there is the opportunity to keep these values the same.

Reply:
Wow, okay, that makes sense!